### ISSUE TITLE
Hyprland keys don't work inside of VirtualBox

### ISSUE THREAD

**Issue by Cornelius-Figgle** [hyprwm/Hyprland#2022]
Running Hyprland in a newly installed Arch guest on a Windows 10 host via VirtualBox

Now I know people say not to test distros etc in a VM, but I would like to be able to play about with it and being as live CDs are, live, you can't save them

Basically, new install with Arch and none of the key bindings are being recognised. At first I thought it was due to the fact by keyboard doesn't have a Windows keys, but rebinding the mod key to Alt doesn't work either. Furthermore, using VirtualBox's soft keyboard and the Windows on-screen keyboard both produce no response

After running `Hyprland` in the TTY, it boots to what looks like the start screen (black with the Hyprland logo), but pressing `MOD+Q` or `MOD+M` does nothing. I have to force shutdown the VM and reboot to exit Hyprland

If it affects it, I installed `hyprland` and `kitty` from `pacman`

- Windows: 10 Pro 21H2
- VirtualBox: 7.0.4
- Arch: 6.2.10-arch1-1
- Hyprland: 0.24.0-1 (pacman)
- GPU: AMD Radeon HD 7400 Series
- Virtual GPU: VMSVGA (3D Virt. Enabled)

**Comment by Person1873** [comment]
hi cornelius
IIRC you need to fully capture your keyboard in VirtualBox.
there should be no reason that a fully captured keyboard would be any different to one that's connected directly to the vm.
does your keyboard work with other wayland based compositors such as sway?

**Comment by Cornelius-Figgle** [comment]
I have keyboard capture on, as per the default for VirtualBox

I have not tried any other compositors, will take a look a sway in a sec

**Comment by Cornelius-Figgle** [comment]
After installing Arch onto a live usb, I managed to get it (mostly) working

```bash
sudo pacman -S hyprland kitty gtk3 seatd foot
sudo systemctl enable seatd.service
sudo systemctl start seatd.service
logout
...
nano ~/.config/hypr/hyprland.conf  # change `MOD+Q` to launch `foot`
Hyprland
```

The above worked, but I couldn't launch `kitty` in Hyprland at all, it simply returned a load of errors on exit (same errors as [here[(https://github.com/hyprwm/Hyprland/issues/1908))

Also I think part of the problem was the VM is bloody slow, so I'm going to up the resource allocation to that

TL;DR: I didn't install everything I needed, and the VM was slow

Sorry for wasting your time ツ
